app/models/card.rb

```diff
@@ -1,2 +1,18 @@
 class Card < ApplicationRecord
+  validates :original_text, :translated_text, :review_date, presence: true
+  validate :texts_are_not_the_same?
+
+  before_validation :default_review_algorithm
```

Comment:
@davetoxa i see, makes sense. i don't think it's ideal to "reset" the review algorithm every single time i update it. it should only set itself upon creation. was that your reasoning why i should use before_create here?